### DISCUSSION TITLE
🗳️ If YOURLS had support for custom themes

### DISCUSSION THREAD

**Discussion by ozh** [YOURLS/YOURLS#2949] [General]
[crossposting from [Twitter](https://twitter.com/ozh/status/1387810235377664002)]

People who code stuff for YOURLS : if YOURLS had support for custom themes, would you prefer it with:
1) templates, a la Twig (left) 
2) PHP+HTML, a la WordPress (right)
Or maybe
3) whichever, would be neat anyway

![image](https://user-images.githubusercontent.com/223647/116611140-512a8000-a936-11eb-97d1-df34b9c81f1e.png)

**Please respond with a "reaction"** (upper right corner of this issue) in order to limit the number of emails sent to everybody watching this repo :)

* Option 1, Twig style : 🚀 
* Option 2, PHP+HTML : 👀 
* Option 3, whichever : 👍

**Comment by mwllgr** [comment]
Just my personal opinion: "Twig style" looks cleaner. I just can't look at a PHP + HTML mixed-mess.

**Comment by clmcavaney** [comment]
I like the Twig template model it provides a cleaner model view control implementation.

**Comment by s22-tech** [comment]
Personally, I would prefer plain PHP with HTML.  Looking at the images above, Twig is not cleaner - there's actually *more* code using Twig - not to mention the ugly double-brackets.  But the big thing for me would be having to learn another "language" when PHP works just fine.  If you do decide to incorporate Twig, please go with option 3 so we're not forced into using it.

Thanks!

**Reply by clmcavaney** [reply]
Yeah. That is fair. Keeping the language ecosystem concise sounds great.

**Reply by f1rf1n** [reply]
Using a template engine makes a lot of sense.

@s22-tech 
Twig is much cleaner. it was actually invented -this century ;-) - for this purpose.  Also the _code is actually quite a bit smaller._ The Twig example has the loop through the data _and_ formats it already! While in the PHP example this is hidden behind the ``yourls_html_print_details()`` function call. I think this would be a lot more than 3 lines, maybe it would almost double the php example in size.

@clmcavaney 
The 'language barrier' is really minimal. But it really helps to separate functionality from presentation. You could basically start using it after reading https://en.wikipedia.org/wiki/Twig_(template_engine)

Finally it is gaining a lot popularity thanks to Symfony amongst others. Personally coming from Laravel and Drupal projects this would be ideal for me (and many others).
